### DISCUSSION TITLE
Wanna be influencers

### DISCUSSION THREAD

**Discussion by wizzymore** [livewire/livewire#4124] [4. General]
There is a guy that posted the following video:
[Laravel Livewire EXTREME security loop hole! ( EXTERNAL: Youtube Link )](https://www.youtube.com/watch?v=kJCGmmSb3m4)

In this video he states that because he put a variable callbed $user_id as a public variable that controls what user information is shown in the component it is now accesible by the frontend and it is a security issue:
```php
public $user_id;
{...}
return view('some_view', [ 'user' => User::find($this->user_id) ]);
```

The first thought when i saw the video was this: "Hmm, maybe he just didn't read the documentation to have an idea of why public variables can be accessed from the fronend javascript"
When i looked at the comment section what surpised me the most was this:
```
For those saying "THIS IS MEANT TO BE THIS WAY" well... the developers I talked to still didn't know about it (actually just one out of ~8 knew about it). So at best it's a documentation issue.
```
His comment that he pinned and acknowledges the fact that he doesn't find any info about this in the documentation and that the devs are not aware that this can happen.

I don't know if he spoke to any devs or not, but, this is stated in the documentation right when the public variables are introduced to the end user:
[https://laravel-livewire.com/docs/2.x/properties#important-notes](https://laravel-livewire.com/docs/2.x/properties#important-notes)

When i posted a comment that specified the fact that this is documented and that the documentation is not a problem he deleted my comment.

I hate this kind of people, using stuff someone else made and distributed for free so everybody can use then people offering to spend time documenting every behaviour of the package and instructing the user what the best practices are and then bashing the devs for something that isn't even true.

**Comment by joshhanley** [comment]
@wizzymore all I can say is good knives are sharp and can do a lot of damage! 😁 

For anyone else that sees this, you should classify `$user_id` as user input, regardless of whether it's changes come from a `wire:model` input or using javascript directly (this is made available so you can use what `wire:model` uses under the hood in Alpine or plain javascript).

As such you should sanitise it and validate it and check current auth user has authorisation, before using it in a query like `User::find($user_id);`

We have been looking into possible solutions to this though, such as being able to mark properties as readonly, so this then wouldn't be possible. But again, you'd still have to intentionally mark a property as `readonly`, out of the box this would still be possible.

Hope this helps!
